<!doctype html>
<html lang="zh" itemscope itemtype="http://schema.org/Person">
<head>
  <meta charset="utf-8">
  <!-- Site Meta Data -->
  <title>GATK4 germline variant calling 分析流程</title>
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="description" content="">
  <meta name="author" content="Wolfson Liu">

  <link rel="shortcut icon" href="https://wolfsonliu.github.io/images/favicon.jpg">

  <!-- schema.org -->
  <meta itemprop="name" content="狼穴">
  <meta itemprop="image" content="">
  <meta itemprop="description" content="">

  <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,600,700' rel='stylesheet' type='text/css'>
  <!-- Style Meta Data -->
  <link rel="stylesheet" href="../../theme/css/style.css" type="text/css" />
  <link rel="stylesheet" href="../../theme/css/pygments.css" type="text/css" />

  <!-- Feed Meta Data -->

  <!-- Twitter Feed -->
  <meta name="twitter:card" content="summary">
  <meta name="twitter:site" content="">
  <meta name="twitter:image" content="">

<meta name="twitter:creator" content="">
<meta name="twitter:url" content="../../archive/2020/gatk4-germline-variant-calling-fen-xi-liu-cheng.html">
<meta name="twitter:title" content="狼穴 ~ GATK4 germline variant calling 分析流程">
<meta name="twitter:description" content="使用 GATK4 call variant.">

<!-- Facebook Meta Data -->
<meta property="og:title" content="狼穴 ~ GATK4 germline variant calling 分析流程" />
<meta property="og:description" content="使用 GATK4 call variant." />
<meta property="og:image" content="" />
</head>

<body>
  <!-- Sidebar -->
  <aside>
    <!--<center><a href="../.."><img id="avatar" src=""></a></center>-->
    <h1>狼穴</h1>
      <p>自然 - 自由</p>
    <br>


    <nav class="nav">
      <ul class="list-bare">
      
          <li><a class="nav__link" href="https://wolfsonliu.github.io/index.html">文章</a></li>
          <li><a class="nav__link" href="https://wolfsonliu.github.io/tag/">标签</a></li>
          <li><a class="nav__link" href="https://wolfsonliu.github.io/category/">类别</a></li>
          <li><a class="nav__link" href="https://wolfsonliu.github.io/about.html">关于</a></li>
          <li><a class="nav__link" href="https://wolfsonliu.github.io/suggestion.html">推荐</a></li>
         
        
      </ul>
    </nav>

    <p class="social">
        <a href="https://github.com/wolfsonliu" target="_blank" ><img src="../../theme/images/icons/github.png"></a>
        <a href="https://www.douban.com/people/77536581/" target="_blank" ><img src="../../theme/images/icons/douban.png"></a>
        <a href="https://ebird.org/ebird/profile/NDY3ODYz" target="_blank" ><img src="../../theme/images/icons/ebird.png"></a>
        <a href="https://zh.wikipedia.org/wiki/User:Wolfsonliu" target="_blank" ><img src="../../theme/images/icons/wikipedia.png"></a>
        <a href="https://orcid.org/0000-0001-6711-4801" target="_blank" ><img src="../../theme/images/icons/orcid.png"></a>
    </p>



  </aside>

  <!-- Content -->
  <article>
<section id="content">
    <article>
        <h2 class="post_title post_detail"><a href="../../archive/2020/gatk4-germline-variant-calling-fen-xi-liu-cheng.html" rel="bookmark" title="Permalink to GATK4 germline variant calling 分析流程">GATK4 germline variant calling 分析流程</a></h2>
        <div class="entry-content blog-post">
            <p>通过对基因组高通量测序数据的分析可以从中获取Genome Varient的信息。这个
过程被称为Call variant，现在已经有很多成熟的软件和流程来完成这个任务。
本笔记主要记录了使用<a href="https://gatk.broadinstitute.org">GATK4</a>软件来分析
Variant的方法。</p>
<h2>软件安装</h2>
<p>为了方便，我直接使用conda的环境来安装Call variant所需要的软件。主要的
软件版本入下：
* GATK4: 4.1.8，主要的分析软件
* Picard: 2.22.8，SAM/BAM处理软件
* BWA: 0.7.17，Mapping 软件</p>
<p>bioconda的channel中包含这些生信分析的软件，bioconda的添加参考<a href="https://wolfsonliu.github.io/archive/2020/python-xue-xi-bi-ji-bioconda-channel.html">介
绍
</a>
。</p>
<p>因为GATK4使用Java 1.8，为了防止和其他软件冲突，最好还是在一个独立的环境中安装。</p>
<div class="highlight"><pre><span></span><code>$ conda create -n gatk gatk4 picard bwa samtools bcftools
</code></pre></div>

<p>这样，我们就创建了一个gatk的环境，其中安装了我们需要的GATK和Picard等软件。</p>
<h2>Call variant 流程</h2>
<p>使用GATK4来进行Variant calling主要参考了<a href="https://github.com/gatk-workflows/gatk4-germline-snps-indels">官方 Germline Variant Calling
流程</a>。
这个分析流程是从已经处理好的BAM文件开始的。生成可以使用的BAM文件则参考
了<a href="https://github.com/gatk-workflows/gatk4-data-processing">官方 Data processing 流
程</a>。</p>
<h3>测序数据比对到基因组</h3>
<p>所得到的原始测序数据是FASTQ格式的，经过预先质量控制（Quality Control）
后就可以比对（Mapping）到基因组。GATK4软件推荐的比对软件是由Heng Li开
发的<a href="http://bio-bwa.sourceforge.net/">BWA</a>。这里用到的参考基因组是由
illumina整理好的hg38/GRCh38
<a href="https://support.illumina.com/sequencing/sequencing_software/igenome.html">iGenome</a>
人类参考基因组。</p>
<div class="highlight"><pre><span></span><code><span class="n">REF_BWA</span><span class="o">=./</span><span class="n">reference</span><span class="o">/</span><span class="n">genome</span><span class="o">.</span><span class="n">fa</span>
<span class="n">THREAD</span><span class="o">=</span><span class="mi">4</span>
<span class="n">FQ1</span><span class="o">=./</span><span class="n">data</span><span class="o">/</span><span class="n">sample</span><span class="o">.</span><span class="mf">1.</span><span class="n">fq</span>
<span class="n">FQ2</span><span class="o">=./</span><span class="n">data</span><span class="o">/</span><span class="n">sample</span><span class="o">.</span><span class="mf">2.</span><span class="n">fq</span>

<span class="n">bwa</span> <span class="n">mem</span> <span class="o">-</span><span class="n">t</span> <span class="o">$</span><span class="p">{</span><span class="n">THREAD</span><span class="p">}</span> <span class="o">$</span><span class="p">{</span><span class="n">REF_BWA</span><span class="p">}</span> <span class="o">$</span><span class="p">{</span><span class="n">FQ1</span><span class="p">}</span> <span class="o">$</span><span class="p">{</span><span class="n">FQ2</span><span class="p">}</span> <span class="o">|</span> \
    <span class="n">samtools</span> <span class="n">view</span> <span class="o">-</span><span class="err">@</span> <span class="o">$</span><span class="p">((</span><span class="n">THREAD</span><span class="o">-</span><span class="mi">1</span><span class="p">))</span> <span class="o">-</span><span class="n">b</span> <span class="o">-</span> <span class="o">&gt;</span> <span class="o">./</span><span class="n">data</span><span class="o">/</span><span class="n">sample</span><span class="o">.</span><span class="n">raw</span><span class="o">.</span><span class="n">bam</span>
</code></pre></div>

<ul>
<li>bwa mem 可以接受双端测序（paired-end），默认会把比对结果输出到标准输
  出（stdout）。</li>
<li>bwa mem 还有一个 <code>-R</code> 参数，可以加上值 '@RG\tID:foo\tSM:bar' 指定读
  取分组，如果设置错误GATK处理会有问题。这个RG可以在比对的时候添加，也
  可以用Picard进行添加，见下面。</li>
<li>使用 samtools view 可以把bwa mem的SAM格式的标准输出转为二进制的BAM格式。</li>
<li><code>$((THREAD-1))</code>是bash变量的数学计算，这里的实际结果是<code>4 - 1=3</code>。
  samtools view 的 <code>-@</code> 参数接受的是增加的线程数，所以这里要减一。</li>
</ul>
<h3>用Picard修改BAM文件的Read Group</h3>
<p>如果在序列比对的时候没有在 bwa mem 中设置Read Group，可以用Picard的
<code>AddOrReplaceReadGroups</code>命令来完成。Read Group主要是表示这些序列是同一
次测序出来的结果，在后面的一些矫正是在一次测序内部进行的，所以需要设置
一下。如果BAM文件中的序列只有单一来源的话，就只要设置这些Read Group完
全一样就可以。下面是通过 Picard 来修改和添加 Read Group。</p>
<div class="highlight"><pre><span></span><code><span class="err">picard AddOrReplaceReadGroups \</span>
<span class="err">    -I ./data/sample.raw.bam \</span>
<span class="err">    -O ./data/sample.AddOrReplaceReadGroups.bam \</span>
<span class="err">    --RGID sample1 \</span>
<span class="err">    --RGLB lib1 \</span>
<span class="err">    --RGPL illumina \</span>
<span class="err">    --RGPU unit1 \</span>
<span class="err">    --RGSM sample1 \</span>
<span class="err">    --VALIDATION_STRINGENCY LENIENT \</span>
<span class="err">    -Xms1g -Xmx10g -XX:ParallelGCThreads=4</span>
</code></pre></div>

<ul>
<li><code>--VALIDATION_STRINGENCY LENIENT</code>这里是为了避免Picard因为检查Header
  不不符合规范而报错终止。</li>
<li><code>-Xms1g</code> 是 Java Virtual Machine的参数，设置最小内存为1G。</li>
<li><code>-Xmx10g</code> 也是 Java Virtual Machine的参数，设置最大内存为10G。</li>
<li><code>-XX:ParallelGCThreads=4</code> 还是 Java Virtual Machine的参数，设置并行
  线程为4。</li>
</ul>
<h3>标注重复序列</h3>
<p>在做基因组测序的时候，序列会因为PCR的缘故而扩增太多，产生的许多重复序
列（Duplicates）会影响到后面计算比例，所以需要区分出那些序列是由于扩增
引起的。这里用Picard的 <code>MarkDuplicates</code> 命令来标注重复序列。这些序列不
会从BAM中被剔除，而只会在
<a href="https://broadinstitute.github.io/picard/explain-flags.html">FLAG</a>处被
标注为重复序列。</p>
<div class="highlight"><pre><span></span><code><span class="err">picard MarkDuplicates \</span>
<span class="err">    -I ./data/sample.AddOrReplaceReadGroups.bam \</span>
<span class="err">    -O ./data/sample.MarkDuplicates.bam \</span>
<span class="err">    -M ./data/sample.MarkDuplicates.matrics \</span>
<span class="err">    -Xms1g -Xmx10g -XX:ParallelGCThreads=4</span>
</code></pre></div>

<h3>对BAM文件进行排序</h3>
<p>在标注完重复序列后，需要对BAM文件按照染色体坐标进行排序。</p>
<div class="highlight"><pre><span></span><code><span class="err">picard SortSam \</span>
<span class="err">    -I ./data/sample.MarkDuplicates.bam \</span>
<span class="err">    -O ./data/sample.SortSam.bam \</span>
<span class="err">    -SO &quot;coordinate&quot; \</span>
<span class="err">    --CREATE_INDEX false \</span>
<span class="err">    --CREATE_MD5_FILE false \</span>
<span class="err">    -Xms1g -Xmx10g -XX:ParallelGCThreads=4</span>
</code></pre></div>

<h3>设置其他SAM标签</h3>
<p>为了方便后面计算，可以设置SAM文件的几个标签：NM、MD和UQ。NM是编辑距离，
即序列可以经过多少编辑变为参考序列。MD是表示错配的字符串。UQ是以Phred
形式表示的匹配到参考序列位置的概率。</p>
<div class="highlight"><pre><span></span><code>REF_FA=./Reference/genome.fa

picard SetNmMdAndUqTags \
    -I ./data/sample.SortSam.bam \
    -O ./data/sample.tagged.bam \
    --CREATE_INDEX true \
    --CREATE_MD5_FILE true \
    --REFERENCE_SEQUENCE <span class="cp">${</span><span class="n">REF_FA</span><span class="cp">}</span> \
    -Xms1g -Xmx10g -XX:ParallelGCThreads=4
</code></pre></div>

<h3>矫正碱基质量得分（Base Quality Score）</h3>
<p>碱基质量是对每个碱基在测序时候估计测序错误概率得出的分值。而测序仪在评
估碱基质量的时候会由于种种原因出现偏差。因为碱基质量得分在后面的Call
Variant过程中会被用做评估碱基的可信程度，所以需要在预处理数据中进行矫
正，避免测序仪的系统偏差。这里利用GATK4 的 <code>BaseRecalibrator</code> 和
<code>ApplyBQSR</code> 命令来进行矫正。其中 <code>Baserecalibrator</code> 通过已知的一些位点
在测序数据中的质量的情况来建立机器学习模型，然后 <code>ApplyBQSR</code> 把模型应
用到所有位点进行矫正。</p>
<div class="highlight"><pre><span></span><code><span class="k">export</span> <span class="n">REF_FA</span><span class="o">=./</span><span class="n">Reference</span><span class="o">/</span><span class="n">genome</span><span class="o">.</span><span class="n">fa</span>
<span class="k">export</span> <span class="n">REF_dbSNP</span><span class="o">=./</span><span class="n">Reference</span><span class="o">/</span><span class="n">Homo_sapiens_assembly38</span><span class="o">.</span><span class="n">dbsnp138</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
<span class="k">export</span> <span class="n">REF_Millsindel</span><span class="o">=./</span><span class="n">Reference</span><span class="o">/</span><span class="n">Mills_and_1000G_gold_standard</span><span class="o">.</span><span class="n">indels</span><span class="o">.</span><span class="n">hg38</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
<span class="k">export</span> <span class="n">REF_knownindel</span><span class="o">=./</span><span class="n">Reference</span><span class="o">/</span><span class="n">Homo_sapiens_assembly38</span><span class="o">.</span><span class="n">known_indels</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>

<span class="n">gatk</span> <span class="n">BaseRecalibrator</span> \
    <span class="o">-</span><span class="n">R</span> <span class="o">$</span><span class="p">{</span><span class="n">REF_FA</span><span class="p">}</span> \
    <span class="o">-</span><span class="n">I</span> <span class="o">./</span><span class="n">data</span><span class="o">/</span><span class="n">sample</span><span class="o">.</span><span class="n">tagged</span><span class="o">.</span><span class="n">bam</span> \
    <span class="o">-</span><span class="n">O</span> <span class="o">./</span><span class="n">data</span><span class="o">/</span><span class="n">sample</span><span class="o">.</span><span class="n">bqsr</span><span class="o">.</span><span class="n">table</span> \
    <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">original</span><span class="o">-</span><span class="n">qualities</span> \
    <span class="o">--</span><span class="n">known</span><span class="o">-</span><span class="n">sites</span> <span class="o">$</span><span class="p">{</span><span class="n">REF_dbSNP</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">known</span><span class="o">-</span><span class="n">sites</span> <span class="o">$</span><span class="p">{</span><span class="n">REF_Millsindel</span><span class="p">}</span> \
    <span class="o">--</span><span class="n">known</span><span class="o">-</span><span class="n">sites</span> <span class="o">$</span><span class="p">{</span><span class="n">REF_knownindel</span><span class="p">}</span>

<span class="n">gatk</span> <span class="n">ApplyBQSR</span> \
    <span class="o">-</span><span class="n">R</span> <span class="o">$</span><span class="p">{</span><span class="n">REF_FA</span><span class="p">}</span> \
    <span class="o">-</span><span class="n">I</span> <span class="o">./</span><span class="n">sample</span><span class="o">.</span><span class="n">tagged</span><span class="o">.</span><span class="n">bam</span> \
    <span class="o">-</span><span class="n">O</span> <span class="o">./</span><span class="n">data</span><span class="o">/</span><span class="n">sample</span><span class="o">.</span><span class="n">bqsr</span><span class="o">.</span><span class="n">bam</span> \
    <span class="o">-</span><span class="n">bqsr</span> <span class="o">./</span><span class="n">data</span><span class="o">/</span><span class="n">sample</span><span class="o">.</span><span class="n">bqsr</span><span class="o">.</span><span class="n">table</span> \
    <span class="o">--</span><span class="k">static</span><span class="o">-</span><span class="n">quantized</span><span class="o">-</span><span class="n">quals</span> <span class="mi">10</span> \
    <span class="o">--</span><span class="k">static</span><span class="o">-</span><span class="n">quantized</span><span class="o">-</span><span class="n">quals</span> <span class="mi">20</span> \
    <span class="o">--</span><span class="k">static</span><span class="o">-</span><span class="n">quantized</span><span class="o">-</span><span class="n">quals</span> <span class="mi">30</span> \
    <span class="o">--</span><span class="n">add</span><span class="o">-</span><span class="n">output</span><span class="o">-</span><span class="n">sam</span><span class="o">-</span><span class="n">program</span><span class="o">-</span><span class="n">record</span> \
    <span class="o">--</span><span class="n">create</span><span class="o">-</span><span class="n">output</span><span class="o">-</span><span class="n">bam</span><span class="o">-</span><span class="n">md5</span> \
    <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">original</span><span class="o">-</span><span class="n">qualities</span>
</code></pre></div>

<ul>
<li>其中所用到的已经知道的VCF文件需要使用 <code>bgzip</code> 来压缩，并利用 <code>tabix</code>
  来建立索引。</li>
</ul>
<h3>Call Variant</h3>
<p>有了经过矫正的BAM文件后,可以通过GATK的<code>HaplotypeCaller</code>命令来获取基本
的VCF。</p>
<div class="highlight"><pre><span></span><code>gatk --java-options &quot;-Xmx10G -XX:ParallelGCThreads=4&quot; \
    HaplotypeCaller \
    -R <span class="cp">${</span><span class="n">REF_FA</span><span class="cp">}</span> \
    -I ./data/sample.bqsr.bam \
    -O ./data/sample.g.vcf.gz \
    -OVI \
    -contamination 0 \
    -G StandardAnnotation \
    -G StandardHCAnnotation \
    -G AS_StandardAnnotation \
    -GQB 10 -GQB 20 -GQB 30 -GQB 40 -GQB 50 -GQB 60 -GQB 70 -GQB 80 -GQB 90 \
    -ERC GVCF \
    -bamout ./data/sample.HaplotypeCaller.bam
</code></pre></div>

<ul>
<li><code>HaplotypeCaller</code> 可以直接输出压缩的 GVCF，并且可以建立
  index（<code>-OVI</code>）。</li>
<li><code>HaplotypeCaller</code> 还可以把Variant附近的序列输出到BAM文件（<code>-bamout</code>）。</li>
</ul>
<h3>对 SNP Variant 得分的矫正</h3>
<p>为了更准确地过滤 VCF，可以对 Variant 的打分进行矫正。对 Variant 的矫正
也类似对碱基的矫正，有两个步骤，需要 GATK 的 <code>VariantRecalibrator</code> 和
<code>ApplyVQSR</code> 两个命令。其中 <code>VariantRecalibrator</code> 也是从已知的 SNP 中建
立模型，而在 <code>ApplyVQSR</code> 中进行矫正。需要注意的是，当前的版本只能每次
单独对 SNP 进行矫正或者对 Indel 进行矫正，而不能同时矫正两个。这里主要
矫正 SNP。</p>
<div class="highlight"><pre><span></span><code>gatk VariantRecalibrator \
    -R <span class="cp">${</span><span class="n">REF_FA</span><span class="cp">}</span> \
    -V ./data/sample.g.vcf.gz \
    --resource:hapmap,known=false,training=true,truth=true,prior=15.0 hapmap_3.3.hg38.sites.vcf.gz \
    --resource:omni,known=false,training=true,truth=false,prior=12.0 1000G_omni2.5.hg38.sites.vcf.gz \
    --resource:1000G,known=false,training=true,truth=false,prior=10.0 1000G_phase1.snps.high_confidence.hg38.vcf.gz \
    --resource:dbsnp,known=true,training=false,truth=false,prior=2.0 Homo_sapiens_assembly38.dbsnp138.vcf.gz \
    -an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR \
    -mode SNP \
    -O ./data/sample.recal \
    --tranches-file ./data/sample.tranches \
    --rscript-file ./data/sample.plots.R

gatk ApplyVQSR \
    -R <span class="cp">${</span><span class="n">REF_FA</span><span class="cp">}</span> \
    -V ./data/sample.g.vcf.gz \
    -O ./data/sample.recalibrated.g.vcf.gz \
    -OVI \
    --truth-sensitivity-filter-level 99.0 \
    --tranches-file ./data/sample.tranches \
    --recal-file ./data/sample.recal \
    -mode SNP
</code></pre></div>

<h2>多任务并行</h2>
<p>在有的时候需要处理的测序样品比较多，想要同时处理多个数据，可以考虑使用<code>xargs</code>来同时运行多个数据。</p>
<div class="highlight"><pre><span></span><code><span class="n">echo</span> <span class="o">-</span><span class="n">e</span> <span class="s2">&quot;sample1 ./bam/sample1.bam&quot;</span> <span class="o">&gt;</span> <span class="n">files</span><span class="o">.</span><span class="n">txt</span>

<span class="k">export</span> <span class="n">DIR_WK</span><span class="o">=$</span><span class="p">(</span><span class="n">pwd</span><span class="p">)</span>

<span class="n">function</span> <span class="n">MarkDuplicates</span> <span class="p">{</span>
    <span class="n">LAB</span><span class="o">=$</span><span class="mi">1</span>
    <span class="n">BAM</span><span class="o">=$</span><span class="mi">2</span>
    <span class="n">OUTDIR</span><span class="o">=$</span><span class="p">{</span><span class="n">DIR_WK</span><span class="p">}</span><span class="o">/</span><span class="n">bam_Sorted</span>
    <span class="n">TMPDIR</span><span class="o">=$</span><span class="p">{</span><span class="n">DIR_WK</span><span class="p">}</span><span class="o">/</span><span class="n">tmp</span>
    <span class="n">mkdir</span> <span class="o">-</span><span class="n">p</span> <span class="o">$</span><span class="p">{</span><span class="n">OUTDIR</span><span class="p">}</span>
    <span class="n">mkdir</span> <span class="o">-</span><span class="n">p</span> <span class="o">$</span><span class="p">{</span><span class="n">TMPDIR</span><span class="p">}</span>
    <span class="n">OUTBAM</span><span class="o">=$</span><span class="p">{</span><span class="n">OUTDIR</span><span class="p">}</span><span class="o">/$</span><span class="p">{</span><span class="n">LAB</span><span class="p">}</span><span class="o">.</span><span class="n">Markduplicates</span><span class="o">.</span><span class="n">bam</span>
    <span class="n">OUTTXT</span><span class="o">=$</span><span class="p">{</span><span class="n">OUTDIR</span><span class="p">}</span><span class="o">/$</span><span class="p">{</span><span class="n">LAB</span><span class="p">}</span><span class="o">.</span><span class="n">MarkDuplicates</span><span class="o">.</span><span class="n">matrics</span>
    <span class="n">picard</span> <span class="n">MarkDuplicates</span> \
           <span class="o">-</span><span class="n">I</span> <span class="o">$</span><span class="p">{</span><span class="n">BAM</span><span class="p">}</span> \
           <span class="o">-</span><span class="n">O</span> <span class="o">$</span><span class="p">{</span><span class="n">OUTBAM</span><span class="p">}</span> \
           <span class="o">-</span><span class="n">M</span> <span class="o">$</span><span class="p">{</span><span class="n">OUTTXT</span><span class="p">}</span> \
           <span class="o">-</span><span class="n">Xms1g</span> <span class="o">-</span><span class="n">Xmx10g</span> <span class="o">-</span><span class="n">XX</span><span class="p">:</span><span class="n">ParallelGCThreads</span><span class="o">=</span><span class="mi">4</span>
<span class="p">}</span>

<span class="k">export</span> <span class="o">-</span><span class="n">f</span> <span class="n">MarkDuplicates</span>

<span class="n">cat</span> <span class="n">files</span><span class="o">.</span><span class="n">txt</span> <span class="o">|</span> <span class="n">xargs</span> <span class="o">-</span><span class="n">P</span> <span class="mi">4</span> <span class="o">-</span><span class="n">n</span> <span class="mi">1</span> <span class="o">-</span><span class="n">I</span> <span class="p">{}</span> <span class="n">bash</span> <span class="o">-</span><span class="n">c</span> <span class="s1">&#39;MarkDuplicates $@&#39;</span> <span class="n">_</span> <span class="p">{}</span>
</code></pre></div>

<ul>
<li><code>files.txt</code>中保存着需要处理的数据，数据有两列，第一列是样品名称，第
  二列是BAM文件的路径。</li>
<li>xargs 的 <code>-P</code>参数指定了并行的任务的数量。</li>
<li>使用 xargs来传参数到自定义的方程需要使用<code>bash -c</code>来运行命令行形式的
  bash脚本，后面的<code>_ {}</code>中的<code>_</code>是用来站第0位的参数的，这样函数中的参数
  <code>$1</code>对应的就是数据文件中的第一列，<code>$2</code>对应的是数据的第二列。</li>
<li>为了使得函数内和<code>bash -c</code>使用变量，就需要<code>export</code>所用到的变量。</li>
</ul>
        </div>
        <div class="post_list">
            <span>By </span>
            <a href="../../author/wolfson-liu.html">@Wolfson Liu</a>
            <span> in </span>
            <span class="post_category"><a href="../../yi-ling-ling-yi.html" rel="bookmark" title="Permalink to 壹零零壹">[ 壹零零壹 ]</a></span>
            <span class="post_date">Mon 27 July 2020</span>
            <div><span>Tags : </span>
                <span><a href="../../tag/bioinformatics.html">#bioinformatics, </a></span>
            </div>

            <div class="entry-social">
                <span class="twitter"><a target="_blank" rel="nofollow" onclick="javascript:window.open(this.href, '', 'menubar=no,toolbar=no,resizable=yes,scrollbars=yes,height=400,width=700');return false;" title="Twitter" href="https://twitter.com/share?url=../../archive/2020/gatk4-germline-variant-calling-fen-xi-liu-cheng.html&text=GATK4 germline variant calling 分析流程&via="><img src="../../theme/images/icons/twitter-s.png"></a></span>

                <span class="gplus"><a target="_blank" title="Google +" href="https://plus.google.com/share?url=../../archive/2020/gatk4-germline-variant-calling-fen-xi-liu-cheng.html&hl=fr" rel="nofollow" onclick="javascript:window.open(this.href, '', 'menubar=no,toolbar=no,resizable=yes,scrollbars=yes,height=450,width=650');return false;"><img src="../../theme/images/icons/google-s.png"></a></span>

                <span class="facebook"><a target="_blank" title="Facebook" rel="nofollow" onclick="javascript:window.open(this.href, '', 'menubar=no,toolbar=no,resizable=yes,scrollbars=yes,height=500,width=700');return false;" href="https://www.facebook.com/sharer.php?u=../../archive/2020/gatk4-germline-variant-calling-fen-xi-liu-cheng.html&t=GATK4 germline variant calling 分析流程"><img src="../../theme/images/icons/facebook-s.png"></a></span>

                <a  target="_blank" title="Linkedin" href="https://www.linkedin.com/shareArticle?mini=true&url=../../archive/2020/gatk4-germline-variant-calling-fen-xi-liu-cheng.html&title=GATK4 germline variant calling 分析流程" rel="nofollow" onclick="javascript:window.open(this.href, '', 'menubar=no,toolbar=no,resizable=yes,scrollbars=yes,height=450,width=650');return false;"><img src="../../theme/images/icons/linkedin-s.png"></a>

                <span class="mail"><a href="mailto:?subject=GATK4 germline variant calling 分析流程&amp;body=Viens découvrir un article à propos de [GATK4 germline variant calling 分析流程] sur le site de Wolfson Liu. ../../archive/2020/gatk4-germline-variant-calling-fen-xi-liu-cheng.html" title="Share by Email" target="_blank"><img src="../../theme/images/icons/mail-s.png"></a></span>
            </div>
        </div>
    </article>
</section>
  </article>

  <!-- Footer -->
  <footer>
    <p>
      Blog powered by <a href="http://getpelican.com/">Pelican</a>, 
      which takes great advantage of <a href="http://python.org">Python</a>.
      Theme <a href="https://github.com/parbhat/pelican-blue">Pelican-Blue</a> by <a href="https://parbhatpuri.com/">@parbhat</a>.
    </p>
  </footer>


</body>
</html>